addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/JeeLinkHandler.java

```
@@ -0,0 +1,94 @@
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.BridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.jeelink.config.JeeLinkConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for a JeeLink USB Receiver thing.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class JeeLinkHandler extends BaseBridgeHandler implements BridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(JeeLinkHandler.class);
+
+    private JeeLinkConnection connection;
+    private ArrayList<JeeLinkReadingConverter<?>> converters;
+
+    public JeeLinkHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            JeeLinkConfig cfg = getConfig().as(JeeLinkConfig.class);
+            connection = AbstractJeeLinkConnection.createFor(cfg);
+            connection.openConnection();
+
+            String initCommands = cfg.initCommands;
+            if (initCommands != null && !initCommands.trim().isEmpty()) {
+                logger.info("Setting init commands for port {}: {}", connection.getPort(), initCommands);
+                connection.setInitCommands(initCommands);
+            }
+
+            converters = SensorDefinition.createConverters(this, cfg.sketchName);
+            for (JeeLinkReadingConverter<?> cnv : converters) {
+                connection.addReadingConverter(cnv);
+            }
+            logger.info("updating JeeLinkHandler for port {} to be ONLINE...", connection.getPort());
+            updateStatus(ThingStatus.ONLINE);
+            logger.info("JeeLinkHandler for port {} is ONLINE.", connection.getPort());
+        } catch (Exception ex) {
```

Do you really need to catch all exceptions or is only a specific subset enough?